src/main/client_context.cpp

```diff
@@ -355,6 +368,18 @@ std::unique_ptr<QueryResult> ClientContext::executeWithParams(PreparedStatement*
         std::string errMsg = exception.what();
         return queryResultWithError(errMsg);
     }
+    // rebind
+    KU_ASSERT(preparedStatement->parsedStatement != nullptr);
+    auto& parameterMap = preparedStatement->parameterMap;
+    preparedStatement->requiredNewTx = transactionContext->hasActiveTransaction() ? false : true;
```

Comment:
Is the right behaviour to not acquire any tx? I thought the right behaviour is to begin a new tx.